### ISSUE TITLE
MNT: Ignore or auto clean-up generated files from testing

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#10037]
### Description

Run tests with `pytest --remote-data`.

### Expected behavior

`git status` to report clean state of the repo upon test completion.

### Actual behavior

```
$ git status
On branch ...
Untracked files:
  (use "git add <file>..." to include in what will be committed)
        compressed_image.fits
        my_image.fits
        new.fits
        new1.fits
        new2.fits
        newtable.fits
        newtable2.fits
        out.fits
        rotation.asdf
        table1.fits
        table2.fits
        table3.fits
        test_file.fits
        test_group.fits
        values.dat
        variable_length_table.fits
```

### System Details

Linux-3.10.0-1062.12.1.el7.x86_64-x86_64-with-redhat-7.7-Maipo
Python 3.7.3 | packaged by conda-forge | (default, Jul  1 2019, 21:52:21) 
[GCC 7.3.0]
Numpy 1.18.1
astropy 4.1.dev756+ge36efa9b9
Scipy 1.3.1

**Comment by saimn** [comment]
Mostly FITS files :scream:

**Comment by saimn** [comment]
That's a problem with docstrings, do we have a solution for running those in a temp directory without polluting the code examples ? (I think this was already discussed somewhere ?)

**Comment by bsipocz** [comment]
cc @astrofrog and @Cadair

**Comment by astrofrog** [comment]
Related issues: https://github.com/astropy/pytest-doctestplus/issues/68 and https://github.com/astropy/astropy/pull/9439

**Comment by pllim** [comment]
> I think this was already discussed somewhere ?

Yes, I remember @mhvk et al. discussed this before but I cannot find this as a standalone issue.

**Comment by bsipocz** [comment]
I can shepherd the doctestplus part of there is a fix pr. Ideally a new release can be done once the 5.4 compatibility is fixed, too.

**Comment by pllim** [comment]
I can have a look but no promises since I never tried to fix doctestplus before.

**Comment by pllim** [comment]
Oh wait, I meant to look at the one that is causing CI failure. But maybe this can be after that fire is put out.

**Comment by pllim** [comment]
Fixed in #9439 (for now).
